adTechModel.py

```diff
@@ -0,0 +1,329 @@
+    bidRequestSizeGB = (
+        getFactOrDefault("bid request size in bytes", facts, defaults, depth - 1) / 1024 / 1024 / 1024
+    )
+    serverToServerEmissionsPerGB = getFactOrDefault(
+        "server to server emissions g per gb", facts, defaults, depth - 1
+    )
+    dataTransferEmissionsPerBidRequest = (
+        externalRequestRatio * bidRequestSizeGB * serverToServerEmissionsPerGB
+    )
+
+    verboseprint(f"{'  ' * (depth - 1)}-------------------------------------------")
+    verboseprint(
+        f"{'  ' * depth}data transfer emissions g per bid request: {dataTransferEmissionsPerBidRequest:.8f} (calculation)"
+    )
+    return dataTransferEmissionsPerBidRequest
+
+
+def getServerEmissionsPerBidRequest(
+    serverAllocation: float, facts: dict[str, float], defaults: dict[str, float], depth: int
+) -> float:
+    serverEmissionsG = (
+        getFactOrDefault("server emissions mt per month", facts, defaults, depth - 1) * 1000000
+    )
+    bidRequests = (
+        getFactOrDefault("bid requests processed billion per month", facts, defaults, depth - 1) * 1000000000
+    )
+    serversProcessingBidRequests = (
+        getFactOrDefault("servers processing bid requests pct", facts, defaults, depth - 1) / 100.0
+    )
+    verboseprint(f"{'  ' * (depth - 1)}-------------------------------------------")
+    serverEmissionsPerBidRequest = (
+        serverAllocation * serverEmissionsG * serversProcessingBidRequests / bidRequests
+    )
+    verboseprint(
+        f"{'  ' * depth}server emissions g per bid request: {serverEmissionsPerBidRequest:.6f} (calculation)"
+    )
+    return serverEmissionsPerBidRequest
+
+
+def getPrimaryEmissionsPerBidRequest(
+    serverAllocation: float,
+    corporateAllocation: float,
+    facts: dict[str, float],
+    defaults: dict[str, float],
+    depth: int,
+) -> float:
+    corporateEmissionsPerBidRequest = getCorporateEmissionsPerBidRequest(
+        corporateAllocation, facts, defaults, depth - 1
+    )
+    dataTransferEmissionsPerBidRequest = getDataTransferEmissionsPerBidRequest(facts, defaults, depth - 1)
+    serverEmissionsPerBidRequest = getServerEmissionsPerBidRequest(
+        serverAllocation, facts, defaults, depth - 1
+    )
+    verboseprint(f"{'  ' * (depth - 1)}-------------------------------------------")
+    primaryEmissionsPerBidRequest = (
+        corporateEmissionsPerBidRequest + dataTransferEmissionsPerBidRequest + serverEmissionsPerBidRequest
+    )
+    verboseprint(
+        f"{'  ' * depth}primary emissions mt per billion bid requests: {primaryEmissionsPerBidRequest:.6f} (calculation)"
+    )
+    return primaryEmissionsPerBidRequest
+
+
+def getSecondaryEmissionsPerBidRequest(
+    syncPartners, facts: dict[str, float], defaults: dict[str, float], depth: int
+) -> float:
+    bidRequestDistributionRatio = getFactOrDefault(
+        "cookie sync distribution ratio", facts, defaults, depth - 1
```

Comment:
Why is this `cookie sync distribution ratio` for `getSecondaryEmissionsPerBidRequest`?